x/claim/keeper/claim_test.go

```diff
@@ -667,3 +669,128 @@ func (suite *KeeperTestSuite) TestGetClaimMetadata() {
+	suite.Require().NoError(err)
+	properClaims := []types.ClaimStatus{{AirdropIdentifier: airdropId, Claimed: false}}
+	suite.Require().Equal(claims, properClaims, "evmos address should have 1 claim record before update")
+
+	// update the address
+	success := suite.app.ClaimKeeper.UpdateAirdropAddress(suite.ctx, evmosParsedAddress, strideAddress, airdropId)
+	suite.Require().True(success, "airdrop update address should succeed")
+
+	// verify that the evmos address CAN NOT claim after the update
+	coins, err = suite.app.ClaimKeeper.GetUserTotalClaimable(suite.ctx, evmosParsedAccAddress, airdropId, false)
+	suite.Require().NoError(err)
+	suite.Require().Equal(coins, emptyCoins, "evmos address should claim 0 coins after update", evmosParsedAccAddress)
+	claims, err = suite.app.ClaimKeeper.GetClaimStatus(suite.ctx, evmosParsedAccAddress)
+	suite.Require().NoError(err)
+	suite.Require().Empty(claims, "evmos address should have 0 claim records after update")
+
+	// verify that the stride address CAN claim after the update
+	coins, err = suite.app.ClaimKeeper.GetUserTotalClaimable(suite.ctx, strideAccAddress, airdropId, false)
+	suite.Require().NoError(err)
+	suite.Require().Equal(coins, sdk.NewCoins(sdk.NewCoin("stake", sdk.NewIntFromUint64(100_000_000))), "stride address should be allowed to claim after update")
+	claims, err = suite.app.ClaimKeeper.GetClaimStatus(suite.ctx, strideAccAddress)
+	suite.Require().NoError(err)
+	suite.Require().Equal(claims, properClaims, "stride address should have 1 claim record after update")
+
+	// claim with the Stride address
+	coins, err = suite.app.ClaimKeeper.ClaimCoinsForAction(suite.ctx, strideAccAddress, types.ACTION_FREE, airdropId)
+	suite.Require().NoError(err)
+	suite.Require().Equal(coins, sdk.NewCoins(sdk.NewCoin("stake", sdk.NewIntFromUint64(20_000_000))), "stride address should be allowed to claim after update")
+	// verify Stride address can't claim again
+	coins, err = suite.app.ClaimKeeper.ClaimCoinsForAction(suite.ctx, strideAccAddress, types.ACTION_FREE, airdropId)
+	suite.Require().NoError(err)
+	suite.Require().Equal(coins, sdk.NewCoins(sdk.NewCoin("stake", sdk.NewIntFromUint64(0))), "can't claim twice after update")
+	// verify Stride address balance went up
+	strideBalance := suite.app.BankKeeper.GetBalance(suite.ctx, strideAccAddress, "stake")
+	suite.Require().Equal(strideBalance.Amount, sdk.NewIntFromUint64(20_000_000), "stride address balance should have increased after claiming")
+}
+
+func (suite *KeeperTestSuite) TestUpdateAirdropAddress_AirdropNotFound() {
+	_, _, evmosParsedAddress, strideAddress := suite.SetupUpdateAirdropAddressChangeTests()
+
+	// update the address
+	success := suite.app.ClaimKeeper.UpdateAirdropAddress(suite.ctx, evmosParsedAddress, strideAddress, "stride")
+	suite.Require().False(success, "airdrop address update shouldn't work with incorrect airdrop id")
+}
+
+func (suite *KeeperTestSuite) TestUpdateAirdropAddress_StrideAddressIncorrect() {
```

Comment:
Can we add a test for the case where airdrop exists, address is parseable, but address does not have a claim record in the airdrop

Reply:
definitely, good call